extensions/amp-analytics/0.1/instrumentation.js

```diff
@@ -94,14 +96,27 @@ class InstrumentationService {
     /** @private {!Observable<!Event>} */
     this.clickObservable_ = new Observable();
 
-    /** @private {!Object<string, !Observable<!AnalyticsEvent>>} */
-    this.observers_ = {};
-
     /** @private {boolean} */
     this.scrollHandlerRegistered_ = false;
 
     /** @private {!Observable<Event>} */
     this.scrollObservable_ = new Observable();
+
+    /** @private {!Object<string, !Observable<!AnalyticsEvent>>} */
+    this.customEventObservers_ = {};
+
+    /**
+     * Some early events have to be buffered because there's no way to predict
+     * how fast all `amp-analytics` elements will be instrumented.
+     * @private {!Array<!AnalyticsEvent>|undefined}
+     */
+    this.customEventBuffer_ = [];
```

Comment:
It'd have to be a map of list. We could do. But in reality, very very few custom events are possible in this case. If you feel strongly, I'll switch.